src/dysh/spectra/spectrum.py

```diff
@@ -129,30 +155,60 @@ def baseline(self, degree, exclude=None, **kwargs):
 
                 Default: no exclude region
 
+            include: list of 2-tuples of int (currently units not supported yet, pending issue 251/260)
+
             model : str
-                One of 'polynomial' or 'chebyshev', Default: 'polynomial'
+                One of 'polynomial' 'chebyshev', 'legendre', or 'hermite'
+                Default: 'polynomial'
             fitter  :  `~astropy.fitting._FitterMeta`
                 The fitter to use. Default: `~astropy.fitter.LinearLSQFitter` (with `calc_uncertaintes=True`).
                 Be care when choosing a different fitter to be sure it is optimized for this problem.
             remove : bool
                 If True, the baseline is removed from the spectrum. Default: False
+            normalize : bool
+                If True, the frequency axis is internally rescaled from 0..nchan-1
+                to avoid roundoff problems (and make the coefficients slightly more
+                understandable). This is usually needed for a polynomial, though overkill
+                for the others who do their own normalization.
+                @todo maybe allow True, False, None; the latter will use True for polynamical, False for others
+                Default: True
 
         """
         # fmt: on
         # @todo: Are exclusion regions OR'd with the existing mask? make that an option?
         kwargs_opts = {
             "remove": False,
+            "normalize": False,
             "model": "polynomial",
             "fitter": LinearLSQFitter(calc_uncertainties=True),
         }
         kwargs_opts.update(kwargs)
 
+        if kwargs_opts["normalize"]:
+            print("Warning:  baseline fit done in channel space")
+            spectral_axis = np.copy(self._spectral_axis.value)
+            nchan = len(spectral_axis)
+            # sadly, there is no single setter for this
+            for i in range(nchan):
+                self._spectral_axis[i] = i * u.Hz  # would like to use "chan" units
```

Comment:
Why not use u.chan ? 
Also you could do this to avoid the loop and not lose the other attributes (_target, _observer, etc) of spectral_axis
self._spectral_axis._value = np.arange(nchan)*u.chan
self._spectral_axis._unit = u.chan

Reply:
doesn't work for me:

```
UnitTypeError: SpectralAxis instances require units equivalent to '(Unit("Hz"), Unit("m"), Unit("J"), Unit("1 / m"), Unit("km / s"), Unit("pix"))', so cannot set it to 'chan'.
```

Reply:
Oh, _value is not the correct attribute -- that does not change

Reply:
I wind up doing a  deepcopy(), that works for now, minus the issue Hz/chan that needs to be resolved.